src/solvers/refinement/string_refinement.cpp

```diff
@@ -1075,7 +1075,10 @@ static optionalt<exprt> get_array(
   if(n>max_string_length)
   {
     stream << "(sr::get_array) long string (size=" << n << ")" << eom;
-    return {};
+    stream << "(sr::get_array) returning a dummy string" << eom;
+    return array_of_exprt(
+      from_integer(CHARACTER_FOR_UNKNOWN, char_type),
+      array_typet(char_type, from_integer(n, index_type)));
```

Comment:
Commit "Return a dummy string when string is too long " - is there a test for this?

Reply:
Do you mean a unit test? Because it would be difficult to test with end-to-end tests.

Reply:
My reading of this (but I could be wrong) is that the behaviour changes with the above patch. If so, it would be nice to have a test that covers this change. If there already is one, then it seems surprising that none of the tests needed to be changed. If there is none, create one - whether a unit test or a regression test is up to you, I don't have sufficient insight.

Reply:
Actually I'm dropping this commit. The problem it was trying to avoid is that sometimes `get` would run out of memory trying to concretize some long strings, so `get` is a more appropriate place to address that (see the commit I added https://github.com/diffblue/cbmc/pull/1873/commits/8d9d9a572f85704a619b4e8d67cd68586f5c291c).
This issue was occurring in one of tests (`jbmc-strings/ValueOf02`) because without the default axioms, non deterministic strings that are not used in the program can be given very big length.